src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -4361,6 +4375,10 @@ lElseClause:
                     expressionType = DirectCast(expression, UnboundLambda).InferredAnonymousDelegate.Key
                     typeList.AddType(expressionType, RequiredConversion.Any, expression)
 
+                ElseIf expressionKind = BoundKind.TupleLiteral Then
+                    expressionType = DirectCast(expression, BoundTupleLiteral).InferredType
```

Comment:
The InferredType is modeled after same thing in ArrayLiteral. It is not optional there as well. It would be an error type if inference is not possible. That would happen, for example, if ValueTuple types are missing. I think I have a testcase for that.